includes/class-gp-translation-helpers.php

```diff
@@ -20,15 +20,27 @@ public static function get_instance() {
 
 	public function __construct() {
 		add_action( 'template_redirect', array( $this, 'register_routes' ), 5 );
-		add_action( 'gp_before_request',    array( $this, 'before_request' ), 10, 2 );
+		add_action( 'gp_before_request', array( $this, 'before_request' ), 10, 2 );
 
 		add_filter(
 			'gp_translation_row_template_more_links',
 			function( $more_links, $project, $locale, $translation_set, $translation ) {
 				$route_translation_helpers = new GP_Route_Translation_Helpers();
-				$permalink = $route_translation_helpers->get_permalink($project->path, $translation->original_id, $translation_set->slug, $translation_set->locale);
-
-				$more_links['discussion'] = '<a href="' . esc_url( $permalink ) . '">Discussion</a>';
+				$permalink                 = $route_translation_helpers->get_permalink( $project->path, $translation->original_id, $translation_set->slug, $translation_set->locale );
+
+				$translation_helper = $this->helpers['discussion'];
+
+				$post_id                  = $translation_helper::get_shadow_post( $translation->original_id );
```

Comment:
I am concerned about the performance of this. This will be queried for each result in a translation set. This will be thousands of queries.

Reply:
I agree on this too, thanks for pointing this out.